korangar_audio/src/lib.rs

```diff
@@ -191,8 +194,57 @@ impl<F: FileLoader> AudioEngine<F> {
             sound_effect_paths: GenerationalSlab::default(),
             sound_effect_track,
         });
+        AudioEngine {
+            engine_context,
+            mute_on_focus_loss: Mutex::new(true),
+        }
+    }
+
+    /// This function set the bool mute_on_focus_loss
+    pub fn set_mute_on_focus_loss(&self, mute_on_focus_loss: bool) {
+        *self.mute_on_focus_loss.lock().unwrap() = mute_on_focus_loss;
+    }
+
+    /// This function activate during audio engine creation
+    pub fn on_create(&self, mute_on_focus_loss: bool) {
+        self.set_mute_on_focus_loss(mute_on_focus_loss);
+    }
+
+    /// This function activate during windows resume state
+    pub fn on_resume(&self) {
+        self.mute(false);
+    }
 
-        AudioEngine { engine_context }
+    /// This function activate during windows suspend state
+    pub fn on_suspended(&self) {
```

Comment:
The audio code doesn't need to know about the focus state of a windows. A "mute" function should be enough.

Reply:
I will create a audio window state to handle the window state interaction.